FccCMakeGuide.md

```diff
@@ -58,11 +56,31 @@ Note that when changing the name of a property of an algorithm or a
 tool, `  make` (and not only `  make packagename` ) needs to be run for
 Gaudi to be aware of the change.
 
-### []() Using an internal library
+### CTest in FCCSW
+
+FCCSW also uses the cmake for integration tests.
+This is described in detail in the documentation page on [adding tests to FCCSW](https://github.com/HEP-FCC/FCCSW/blob/master/doc/AddingTestsToFCCSW.md).
+
+### Using an internal library
+
+Once Gaudi is notified that a certain subdirectory is needed by invoking `gaudi_depends_on_subdir`, internal libraries defined in this subdirectory can be used by simply adding them to the list of `INCLUDE_DIRS` and `LINK_LIBRARIES`. An example would be the way the internal library `DetCommon` is used by the module [`Test/TestGeometryPlugins`](https://github.com/HEP-FCC/FCCSW/blob/master/Test/TestGeometry/CMakeLists.txt) in FCCSW.
+
+
+
+### Building a new Gaudi module
+
+The best way is to look at existing modules in FCCSW for inspiration. The syntax to declare the module [`TestGeometryPlugins`](https://github.com/HEP-FCC/FCCSW/blob/master/Test/TestGeometry/CMakeLists.txt), for example, is:
```

Comment:
Can you prefix this with a definition of what a module is? I.e. when do I `gaudi_add_module` and when do I do `gaudi_add_library`?